### PR TITLE
General bug fixes and some user experience improvements

### LINKED ISSUE #1121: Fix/improve user path
**Chats**
Currently if a user go to "Space A" and then to the spaces page and click on a Sub space (Space B), when the user then click on the back button, the user is sent back to the home page. 

Instead, the user should be directed back to the previous page. In this example it means, that when the user click on the back button from Space B, the user should be directed back to Space A.

**Chats**
The same should be for chats, meaning if a user click and enter a chat from a space and the user click on the back button, the user should be directed back to the space, not the chat overview. The user only be directed back to the chat overview, if the user entered the chat from the chat overview.

### LINKED ISSUE #1146: remove the option to create pins and events for users who dont have permission.
Its confusing and frustrating that you as a user can go all the way to the point that its loading in creation process, but then shows error.

Image

### PR DESCRIPTION
**Pull request by kumarpalsinh25**
- Changes in back navigation for Sub-spaces, fixes #1121 
- Managed post ping/event option based on the permission, fixes #1146 
- Minor correction on routing: After redacting of pin, it should redirect back to pin listing instead of going to event list
- Added support for full screen image view in chat, ref #1115

### CODE REVIEW

File: .changes/1176-general-fixes.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+- [bugfix] : Improved back navigation between sub-spaces and space
+- [bugfix] : Ping/Event creation is now based on the required permission
```
Comment (gnunicorn) [inline .changes/1176-general-fixes.md:2]:
```suggestion
- [bugfix] : Pin/Event creation is now based on the required permission
```

File: app/lib/features/chat/widgets/image_message_builder.dart [hunk 2]
```diff
@@ -46,48 +47,64 @@ class _ImageMessageBuilderState extends State<ImageMessageBuilder> {
     }
   }
 
+  void showFullScreenImage(Uint8List data) {
+    showAdaptiveDialog(
+      context: context,
+      barrierDismissible: false,
+      useRootNavigator: false,
+      builder: (ctx) => ImageDialog(
+        title: 'Image',
```
Comment (gnunicorn) [inline app/lib/features/chat/widgets/image_message_builder.dart:56]: shouldn't the image-message have a name as well?
```suggestion
        title: widget.message.name(),
```

File: app/lib/features/space/pages/pins_page.dart [hunk 3]
```diff
@@ -19,81 +19,106 @@ class SpacePinsPage extends ConsumerWidget {
 
   @override
   Widget build(BuildContext context, WidgetRef ref) {
+    final spaceRelationsOverview =
```
Comment (gnunicorn) [inline app/lib/features/space/pages/pins_page.dart:22]: that's not the right provider for this. there is a provider that directly yields the space profile with the membership. We don't need that entire relationship thing here.

File: app/lib/features/space/pages/pins_page.dart [hunk 4]
```diff
@@ -19,81 +19,106 @@ class SpacePinsPage extends ConsumerWidget {
...
+                            color: Theme.of(context).colorScheme.neutral5,
+                          ),
+                          iconSize: 28,
+                          color: Theme.of(context).colorScheme.surface,
+                          onPressed: () => context.pushNamed(
+                            Routes.actionAddPin.name,
+                            queryParameters: {'spaceId': spaceIdOrAlias},
+                          ),
+                        ),
+                      ),
```
Comment (gnunicorn) [inline app/lib/features/space/pages/pins_page.dart:73]: If we do that entire spaceRelationsOverview only to get the canPostPin figured out here, we probably do not want to wait for that to show the pins. Instead I'd suggest to replace the Visibility with a Consumer or a widget specific that then does the provider call ..

looking at that and how often we have that pattern we might want to create a common reusable Widget `UserCan(roomId, canString, child: IconButton( ... ) )` that does a `roomProviderMembership`-lookup, checks for that `canString` and if that comes with false, renders a `SizedBox.shrink()`.

**Review by gnunicorn — APPROVED** [review]
Some improvement suggestions.

File: app/lib/features/space/pages/pins_page.dart (hunk 4 above)
Comment (kumarpalsinh25) [reply app/lib/features/space/pages/pins_page.dart:73]: Good notice.
I will create reusable components for this.

**Review by gtalha07 — APPROVED** [review]
LGTM!

**Pull request merged**